CMakeLists.txt

```diff
@@ -121,8 +122,11 @@ endif()
 set(TILEDB_LIB_DEPENDENCIES ${TILEDB_LIB_DEPENDENCIES} pthread)
 
 # Add HDFS support
-add_definitions(-DUSE_HDFS)
-set(TILEDB_LIB_DEPENDENCIES ${TILEDB_LIB_DEPENDENCIES} ${CMAKE_DL_LIBS})
+set(HDFS_SOURCE_DIR "${CMAKE_SOURCE_DIR}/deps/HDFSWrapper/hadoop-hdfs-native")
+find_package(HDFS REQUIRED)
+include_directories(${HDFS_INCLUDE_DIR})
+set_source_files_properties(${HDFS_OBJS} PROPERTIES GENERATED TRUE)
+set(TILEDB_LIB_DEPENDENCIES ${TILEDB_LIB_DEPENDENCIES} ${JAVA_JVM_LIBRARY} ${CMAKE_DL_LIBS})
```

Comment:
Since you are not  really compiling the HDFS sources independently into a library (no top level CMakeLists.txt file in the repo), can you just use add_directory() rather than using ExternalProject? This would avoid a lot of gymnastics with respect to stuff like using set_source_file_properties()

Reply:
There is a CMakeLists.txt in HDFS_SOURCE_DIR that gets invoked with ExternalProject. Yes, the set_source_file_properties is a little convoluted, but we are keeping the builds isolated.